File: frontend/src/types/segment.ts
export interface SegmentRule {
  field: string
  operator: string
  value: string
}

export interface SegmentDetail {
  id: string
  name: string
  description: string
  icon: string
  color: string
  customersCount: number
  rules: SegmentRule[]
  enabledSkills: string[]
  recommendedProducts: string[]
  activatedAgents: string[]
}

File: backend/app/schemas/segment.py
from pydantic import BaseModel


class SegmentRule(BaseModel):
    field: str
    operator: str
    value: str


class Segment(BaseModel):
    id: str
    name: str
    description: str
    icon: str
    color: str
    customersCount: int
    rules: list[SegmentRule]
    enabledSkills: list[str]
    recommendedProducts: list[str]
    activatedAgents: list[str]
